### PR TITLE
Initial security section

### PR DESCRIPTION
**Pull request by Deep-Six**
## Overview
Initial Security Structure 

### Features

- add security section 

---

#### Meta

- [ ] provide a descriptive topic
- [ ] provide an overview of contribution
- [ ] no sensitive content included, such as:
  - security & privacy policy violating content
  - content considered competitive intelligence
  - keys, tokens or credentials
- [ ] documentation format follows [this template][template]
- [ ] fork is up to date ([see this guide from github](https://help.github.com/articles/syncing-a-fork/))
- [ ] "work in progress" commits are squashed ([see "Squashing Commits"](https://git-scm.com/book/id/v2/Git-Tools-Rewriting-History))
- [ ] commits follow the [Karma][karma-format] or [Commitizen](https://www.npmjs.com/package/commitizen) format

[template]: ../.template.md
[karma-format]: https://karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: security/ASVS.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Why
+OWASP's Application Security Verification Standard [OWASP
```
Comment (ahmadnassri) [inline security/ASVS.md]: this is broken markdown

File: security/ASVS.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ASVS](https://www.owasp.org/index.php/Category:OWASP_Application_Security_Verification_Standard_Project) is a prescriptive controls matrix which offers a certification level based on incrementally more strict control objectives. Agreeing to a certification can enable your team to have a rallying point over what controls are necessary within your project. These controls can be used to craft testable user stories and test criteria developed to authored as part of a TDD practice. The levels (1-3) increase in breadth and depth from 1 - Opportunistic, 2 - Standard and 3 Advanced. Level 0 is able to be customised, and is being explored by TELUS Digital Security team. Having a standardised methodology for evaluating the security posture of projects will enable a more consistent security practice, regardless of who is involved. Additionally, having a standard will enable a common currency for the interchange of ideas and design patterns.
+
+## What
+Agreeing to an ASVS Security Level  objective at the inception of your project can
```
Comment (ahmadnassri) [inline security/ASVS.md]: fix your spacing and line breaks please

File: security/ASVS.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ensure all the security concerns are flushed out over the course of your
+project. User stories and Evil User stories, based off the control bjectives (and violation of control objectives) can ensure the requisite
```
Comment (ahmadnassri) [inline security/ASVS.md]: link to a reference on what Evil User Stories are

File: security/ASVS.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+team as a common objective.
+Wherever possible these stories should have a testable element with acceptance
+criteria.
+## How
+With your team, identify the control objective your project is intent on
+achieving  (1-3). Thereafter use this level and it's prescribed controls to
+craft user stories and evil user stories ensuring the prescribed controls are
+serviced and testable. 
+## Who
+Product Owners should consult with their assigned security team resource to
```
Comment (ahmadnassri) [inline security/ASVS.md]: just use `@everyone` here as the "who"

File: security/ASVS.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+## Why
+OWASP's Application Security Verification Standard [OWASP
+ASVS](https://www.owasp.org/index.php/Category:OWASP_Application_Security_Verification_Standard_Project) is a prescriptive controls matrix which offers a certification level based on incrementally more strict control objectives. Agreeing to a certification can enable your team to have a rallying point over what controls are necessary within your project. These controls can be used to craft testable user stories and test criteria developed to authored as part of a TDD practice. The levels (1-3) increase in breadth and depth from 1 - Opportunistic, 2 - Standard and 3 Advanced. Level 0 is able to be customised, and is being explored by TELUS Digital Security team. Having a standardised methodology for evaluating the security posture of projects will enable a more consistent security practice, regardless of who is involved. Additionally, having a standard will enable a common currency for the interchange of ideas and design patterns.
+
+## What
```
Comment (ahmadnassri) [inline security/ASVS.md]: the what section seems to talk a bit about "how to approach this (describing stories)"

File: security/ASVS.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Agreeing to an ASVS Security Level  objective at the inception of your project can
+ensure all the security concerns are flushed out over the course of your
+project. User stories and Evil User stories, based off the control bjectives (and violation of control objectives) can ensure the requisite
+security concern space is entirely serviced, and no gaps exist. Formalizing
+this discussion enables a common understanding that can be socialized with the
+team as a common objective.
+Wherever possible these stories should have a testable element with acceptance
+criteria.
+## How
```
Comment (ahmadnassri) [inline security/ASVS.md]: the how section should be more descriptive and provide a usable and repeatable formula / playbook / steps that team members can use here to achieve the desired outcome

File: security/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
+![Reference Architecture Wiki Logo](logo.png "Reference Architecture Wiki")
```
Comment (ahmadnassri) [inline security/README.md]: don't need the logo here

File: security/README.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+---
+> **Note**: This is a work in progress, you are welcome to contribute and suggest changes, please follow the [contribution guidelines](.github/CONTRIBUTING.md)
```
Comment (ahmadnassri) [inline security/README.md]: you shouldn't have to include this note, the entire repo is always going to be a work in progress and contribution is open by default

File: security/README.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+## Why
+
+Security is everyone's responsibility, by including security mindfulness early in the development process, so security concerns are identified and remediated ideally before they are put into production. This avoids the costly (both in terms of
+time and money) refactoring of a project later in the development cycle. Moreover, our customers expect the Confidentiality, Identity and Availability of our products and services to be an intrinsic value.
```
Comment (ahmadnassri) [inline security/README.md]: line break at odd location

File: security/README.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+## What
+
+The following sections and placeholder articles are the the bare minimum your project should consider as part of it's "Secure by Design" development.
```
Comment (ahmadnassri) [inline security/README.md]: what sections? please follow a hierarchical header structure here

File: security/README.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+## How
+
+First and foremost, application security is a relatively new discipline, the humility and wisdom to ask for help is a great start. There's a lot of moving parts and new threats are a given. Connecting with your outcome team's assigned @security resource can help you overcome some initial hurdles.
+
+## Who
+
+As mentioned above, security is everyone's responsibility our customers are dependent upon our decision making to ensure the custody of their data. The @security team is here to help with guidance, find answers  and to coach on secure practices.
```
Comment (ahmadnassri) [inline security/README.md]: if it's mentioned above, there is no purpose to repeat it.

Comment (ahmadnassri) [reply security/README.md]: missing punctuation

Comment (ahmadnassri) [reply security/README.md]: see previous note on "how" section, need to have actionable, repeatable and tasks / playbook

File: security/ASVS.md (hunk 1 above)
Comment (Deep-Six) [reply security/ASVS.md]: ok, will fix was using a copy of another RA section as a template.

File: security/ASVS.md (hunk 5 above)
Comment (Deep-Six) [reply security/ASVS.md]: Might need more detail lets chat (i'll set up time with you)

File: security/ASVS.md (hunk 2 above)
Comment (Deep-Six) [reply security/ASVS.md]: ok, will fix was using a copy of another RA section as a template.

File: security/README.md (hunk 7 above)
Comment (Deep-Six) [reply security/README.md]: ok, will fix was using a copy of another RA section as a template.

File: security/README.md (hunk 8 above)
Comment (Deep-Six) [reply security/README.md]: ok, will fix was using a copy of another RA section as a template.

File: security/ASVS.md (hunk 1 above)
Comment (Deep-Six) [reply security/ASVS.md]: @ahmadnassri  Fixed

File: security/ASVS.md (hunk 5 above)
Comment (Deep-Six) [reply security/ASVS.md]: @ahmadnassri Fixed

File: security/ASVS.md (hunk 3 above)
Comment (Deep-Six) [reply security/ASVS.md]: Done

File: security/ASVS.md (hunk 6 above)
Comment (Deep-Six) [reply security/ASVS.md]: Done, but needs incremental improvement.

File: security/ASVS.md (hunk 4 above)
Comment (Deep-Six) [reply security/ASVS.md]: Done

File: security/README.md (hunk 7 above)
Comment (Deep-Six) [reply security/README.md]: Done

File: security/README.md (hunk 9 above)
Comment (Deep-Six) [reply security/README.md]: Fixed

File: security/README.md (hunk 10 above)
Comment (Deep-Six) [reply security/README.md]: Revised

**Comment by crimeminister** [comment]
Pinging @alexandarp for insight!

**Comment by crimeminister** [comment]
@alexandarp Does the section on threat modelling seem worth splitting out into a separate contribution?

**Comment by alexandarp** [comment]
@crimeminister Yes.

**Pull request closed without merging**